src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../utils/api";
+import SingleArticleCard from "./SingleArticleCard";
+
+const SingleArticle = ({ articles, setArticles }) => {
+  const { articleid } = useParams();
+  useEffect(() => {
+    getArticleById(articleid).then(({ data: { article } }) => {
+      setArticles(article);
```

Comment:
You have re-used your [articles, setArticles] state here which is working fine but it would be best practice to have a new state that handles the value of your single article e.g. [singleArticle, setSingleArticle] as they are technically two different things in your app.